amr-wind/utilities/sampling/PlaneSampler.cpp

```diff
@@ -58,6 +58,40 @@ void PlaneSampler::check_bounds()
     const auto* prob_lo = m_sim.mesh().Geom(lev).ProbLo();
     const auto* prob_hi = m_sim.mesh().Geom(lev).ProbHi();
 
+    // Do the checks based on a small fraction of the minimum dx
+    const auto& fine_geom = m_sim.mesh().Geom(m_sim.mesh().finestLevel());
+    amrex::Real min_dx = fine_geom.CellSize(0);
+    for (int d = 1; d < AMREX_SPACEDIM; ++d) {
+        min_dx = std::min(fine_geom.CellSize(d), min_dx);
+    }
+    const auto tol = std::max(1e-10 * min_dx, bounds_tol);
+
+    // First fix the origin so that it is within bounds, if it is close enough
+    for (int d = 0; d < AMREX_SPACEDIM; ++d) {
+        if (amrex::Math::abs(m_origin[d] - prob_lo[d]) < tol) {
+            m_origin[d] = prob_lo[d] + tol;
```

Comment:
Curious if we want `(1+tol) * prob_lo` and `(1-tol) * prob_hi` instead of `+-tol` directly. 

(Given the tests, I'm guessing not, but if we get to `numeric_limits::epsilon` instead of `1e-10*dx`, the `tol` wouldn't add anything right?)

Reply:
Oh nvm, just saw what the actual `if` condition is checking.